lib/authentication-context.js

```diff
@@ -248,7 +248,8 @@ AuthenticationContext.prototype.acquireToken = function(resource, userId, client
  * @param {string}   clientId                            The OAuth client id of the calling application.
  * @param {AcquireTokenCallback}   callback              The callback function.
  */
-AuthenticationContext.prototype.acquireTokenWithUsernamePassword = function(resource, username, password, clientId, callback)  {
+AuthenticationContext.prototype.acquireTokenWithUsernamePassword = function(resource, username, password, clientId, clientSecret, callback)  {
```

Comment:
@phedinkus This is a breaking change.  I think we are better off adding a whole new function.  You could potentially make the implementation of this one a special case of the new one.

Reply:
It's not just this example that is broken, but ADAL for Ruby and it's user_credentials example (which also uses the graph API) has the same problem. What's the point of a sample that is so heavily broken?